### PR TITLE
`@guild-docs/algolia`

### PR DESCRIPTION
**Pull request by charlypoly**

### CODE REVIEW

File: packages/algolia/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion, no-else-return */
+import sortBy from 'lodash/sortBy.js';
+import reduce from 'lodash/reduce.js';
+import isString from 'lodash/isString.js';
+import isArray from 'lodash/isArray.js';
+import flatten from 'lodash/flatten.js';
+import each from 'lodash/each.js';
```
Comment (B2o5T) [inline packages/algolia/src/index.ts:7]:
```suggestion
import { sortBy, reduce, isString, isArray, flatten, each } from 'lodash';
```

File: packages/algolia/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+const extractToC = (content: string) => {
+  const slugger = new GithubSlugger();
+
+  const lines = content.split('\n');
+
+  let isCodeBlock = false;
+  let currentDepth = 0;
+  let currentParent: AlgoliaSearchItemTOC | undefined;
+
+  const slugs = reduce<string, AlgoliaSearchItemTOC[]>(
```
Comment (B2o5T) [inline packages/algolia/src/index.ts:26]: use native `reduce` function

File: packages/algolia/src/index.ts (hunk 1 above)
Comment (charlypoly) [reply packages/algolia/src/index.ts:7]: Does not work in ESM, that's why I've put it this way

Comment (B2o5T) [reply packages/algolia/src/index.ts:7]: can you install only functions that don't exist in native js to make bundler smaller, I guess:
```
reduce => Array.reduce
isString => typeof foo === 'string'
isArray => Array.isArray
flatten => Array.flat
each => Array.forEach or for (const [key,value] of Object.entries)
```

File: packages/algolia/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        })
+        .then(() => {
+          index
+            .saveObjects(objects)
+            .then(({ objectIDs }) => {
+              console.log(objectIDs);
+            })
+            .catch(console.error);
+        })
+        .catch(console.error);
```
Comment (B2o5T) [inline packages/algolia/src/index.ts:263]:
```suggestion
        .then(() => index.saveObjects(objects))
        .then(({ objectIDs }) => {
          console.log(objectIDs);
        })
        .catch(console.error);
```

File: packages/algolia/src/index.ts (hunk 2 above)
Comment (charlypoly) [reply packages/algolia/src/index.ts]: Fixed in 3b5b77a

**Pull request merged**